Core/src/main/java/org/tribuo/protos/ProtoUtil.java

```diff
@@ -0,0 +1,474 @@
+
+                    field.setAccessible(false);
+                }
+                serializedClassBuilderClass.getMethod("setSerializedData", com.google.protobuf.Any.class).invoke(serializedClassBuilder, Any.pack(serializedDataBuilder.build()));
+            }
+            return (SERIALIZED_CLASS) serializedClassBuilder.build();
+        } catch (InvocationTargetException | IllegalAccessException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * If this class is annotated with {@link ProtoSerializableClass} returns the
+     * version number, otherwise returns -1.
+     *
+     * @param clazz The class to check.
+     * @return The version number, or -1 if it does not use the annotation.
+     */
+    public static int getCurrentVersion(Class<? extends ProtoSerializable<?>> clazz) {
+        ProtoSerializableClass ann = clazz.getAnnotation(ProtoSerializableClass.class);
+        if (ann != null) {
+            return ann.version();
+        } else {
+            return -1;
+        }
+    }
+
+    static <SERIALIZED_CLASS extends Message, PROTO_SERIALIZABLE extends ProtoSerializable<SERIALIZED_CLASS>> Class<SERIALIZED_CLASS> getSerializedClass(PROTO_SERIALIZABLE protoSerializable) {
+        @SuppressWarnings("unchecked")
+        Class<SERIALIZED_CLASS> serializedClass = (Class<SERIALIZED_CLASS>) resolveTypeParameter(ProtoSerializable.class, protoSerializable.getClass(), ProtoSerializable.class.getTypeParameters()[0]);
+        if (serializedClass != null) {
+            return serializedClass;
+        }
+        String tpName = ProtoSerializable.class.getTypeParameters()[0].getName();
+        throw new IllegalArgumentException("unable to resolve type parameter '" + tpName + "' in ProtoSerializable<" + tpName + "> for class " + protoSerializable.getClass().getName());
+    }
+
+    private static List<Object> convertValues(Map<?, ?> obj) {
+        List<Object> values = new ArrayList<>();
+        for (Object value : obj.values()) {
+            values.add(convert(value));
+        }
+        return values;
+    }
+
+    private static Object convert(Object obj) {
+        if (obj instanceof ProtoSerializable) {
+            return ((ProtoSerializable<?>) obj).serialize();
+        } else if (obj instanceof MutableLong) {
+            return ((MutableLong) obj).longValue();
+        } else if (obj instanceof MutableDouble) {
+            return ((MutableDouble) obj).doubleValue();
+        } else if (obj.getClass().isEnum()) {
+            return ((Enum<?>) obj).name();
+        } else if (obj instanceof int[]) {
+            int[] t = (int[]) obj;
+            return Arrays.stream(t).boxed().collect(Collectors.toList());
+        } else if (obj instanceof long[]) {
+            long[] t = (long[]) obj;
+            return Arrays.stream(t).boxed().collect(Collectors.toList());
+        } else if (obj instanceof float[]) {
+            float[] t = (float[]) obj;
```

Comment:
Is there a reason the logic is different here?

Reply:
There's no `FloatStream` so you can't do `Arrays.stream`.